platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronizedInstallerImpl.java

```diff
@@ -595,4 +595,28 @@ public boolean isUpdated() {
       return updated;
     }
   }
+
+  private static class BundleStates {
+    private Set<Bundle> unavailableBundles;
+    private Set<Bundle> failedBundles;
+
+    BundleStates(Set<Bundle> unavailableBundles, Set<Bundle> failedBundles) {
+      this.unavailableBundles =
+          unavailableBundles == null ? Collections.emptySet() : unavailableBundles;
+      this.failedBundles = failedBundles == null ? Collections.emptySet() : unavailableBundles;
```

Comment:
❗️ Should be setting to `failedBundles` in the second half of the ternary.

If you make this class `package-private` for testing, you can add a test for the four conditions: null or non-null inputs for the two collections and confirm for that.

Reply:
Good catch! I added test for the `getUnavailableBundles` method instead since it will test this class plus that methods logic. I actually found an additional bug, thanks for recommending additional testing here.